### ISSUE TITLE
warn about shadowed format args?

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#8589]
### What it does

Warn about format args that declared for the same format string.

````rust
#![warn(clippy::shadow_reuse, clippy::shadow_same, clippy::shadow_unrelated)]
pub fn main() {
    let x = 2;
    println!("{x} {y}", x = x*x, y=x);
}
````

one might think that this could print `4 4` but it will actually print `4 2` because the `y=x` binding binds the `let x` variable and not the `x =x*x` that we declared in the format string previously

### Lint Name

shadow_format maybe?

### Category

style

### Example

```rust
#![warn(clippy::shadow_reuse, clippy::shadow_same, clippy::shadow_unrelated)]
pub fn main() {
    let x = 2;
    println!("{x} {y}", x = x*x, y=x);
}
```

Could be written as:

```rust
#![warn(clippy::shadow_reuse, clippy::shadow_same, clippy::shadow_unrelated)]
pub fn main() {
    let x = 2;
    println!("{z} {y}", z = x*x, y=x);
}
```
